packages/template-retail-react-app/app/pages/product-list/index.jsx

```diff
@@ -546,34 +546,32 @@ const ProductList = (props) => {
 
 ProductList.getTemplateName = () => 'product-list'
 
-ProductList.shouldGetProps = ({previousLocation, location}) =>
+ProductList.shouldGetProps = ({ previousLocation, location }) =>
     !previousLocation ||
     previousLocation.pathname !== location.pathname ||
     previousLocation.search !== location.search
 
-ProductList.getProps = async ({res, params, location, api}) => {
-    const {categoryId} = params
+ProductList.getProps = async ({ res, params, location, api }) => {
+    const { categoryId } = params
     const urlParams = new URLSearchParams(location.search)
-    let searchQuery = urlParams.get('q')
-    let isSearch = false
 
-    if (searchQuery) {
-        isSearch = true
-    }
-    // In case somebody navigates to /search without a param
-    if (!categoryId && !isSearch) {
-        // We will simulate search for empty string
-        return {searchQuery: ' ', productSearchResult: {}}
+    const searchQuery = urlParams.get('q')
+    const isSearch = !!searchQuery
+    const isEmptySearch = !categoryId && !isSearch
+    if (isEmptySearch) {
+        return { searchQuery: ' ', productSearchResult: {} }
     }
 
     const searchParams = parseSearchParams(location.search, false)
 
-    if (!searchParams.refine.includes(`cgid=${categoryId}`) && categoryId) {
+    const isCategorySearch = !searchParams.refine.includes(`cgid=${categoryId}`) && categoryId
+    if (isCategorySearch) {
         searchParams.refine.push(`cgid=${categoryId}`)
     }
 
+    // @@@ Disabled. Search _ALL_ products instead!
     // only search master products
-    searchParams.refine.push('htype=master')
+    // searchParams.refine.push('htype=master')
```

Comment:
The decision to only show "master" products is arbitrary and won't work for many catalogs.